nengo/tests/test_pytest.py

```diff
@@ -10,31 +12,91 @@ def test_seed_fixture(seed):
     assert i == 1832276344
 
 
-def test_unsupported(testdir):
+@pytest.mark.parametrize("xfail", (True, False))
+def test_unsupported(xfail, testdir):
+    testdir.makefile(".py", test_file="""
+        import pytest
+
+        @pytest.mark.parametrize("param", (True, False))
+        def test_unsupported(param):
+            assert param
+
+        @pytest.mark.parametrize("param", (True, False))
+        def test_unsupported_all(param):
+            assert False
+
+        def test_supported():
+            assert True
+        """)
+
+    testdir.makefile(".ini", pytest="""
+        [pytest]
+        nengo_test_unsupported =
+            test_file.py:test_unsupported[False]
+                "One unsupported param
+                with multiline comment"
+            test_file.py:test_unsupported_all*
+                "Two unsupported params
+                with multiline comment"
+        """)
+
+    testdir.makefile(".py", conftest="""
+        from nengo.conftest import pytest_runtest_setup, pytest_configure
+        """)
+
+    args = "-p nengo.tests.options -rsx".split()
+    if xfail:
+        args.append("--unsupported")
+    output = testdir.runpytest(*args)
+
+    output.stdout.fnmatch_lines_random([
+        "*One unsupported param with multiline comment",
+        "*Two unsupported params with multiline comment",
+    ])
+
+    outcomes = output.parseoutcomes()
+    if xfail:
+        assert outcomes["xfailed"] == 3
+        assert "skipped" not in outcomes
+    else:
+        assert outcomes["skipped"] == 3
+        assert "xfailed" not in outcomes
+    assert "failed" not in outcomes
+    assert outcomes["passed"] == 2
+
+
+def test_pyargs(testdir):
+    # create a simulator that immediately fails (we don't actually want
```

Comment:
Note that the point of this simulator was to immediately fail (it isn't an actual simulator, so it will produce an error as soon as the test tries to create a simulator).  It's not just that it didn't support the tests.

Reply:
"fails" just seemed ambiguous to me. Fails at what? It's not like it raises an exception, so it's not guaranteed to fail anything. Also, if we're skipping instead of xfailing below, then it doesn't actually fail. Though you make a good point below that this isn't even necessary if we skip instead of xfail.